Core/Loaders.cpp

```diff
@@ -259,11 +251,12 @@ bool LoadFile(FileLoader **fileLoaderPtr, std::string *error_string) {
 					coreState = CORE_BOOT_ERROR;
 					return false;
 				}
-				std::string path = fileLoader->GetPath();
-				size_t pos = path.find("PSP/GAME/");
+
+				std::string dir = fileLoader->GetPath().GetDirectory();
+				size_t pos = dir.find("PSP/GAME/");
 				if (pos != std::string::npos) {
-					path = ResolvePBPDirectory(path);
-					pspFileSystem.SetStartingDirectory("ms0:/" + path.substr(pos));
+					dir = ResolvePBPDirectory(Path(dir)).ToString();
+					pspFileSystem.SetStartingDirectory("ms0:/" + dir.substr(pos));
```

Comment:
It will have correct slashes because we use / slashes internally, and don't call .ToWString(). That said, there is a bit of an unholy mix of path types here.